src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1100,10 +1100,17 @@ var ReactCompositeComponent = {
    * @final
    * @private
    */
-  attachRef: function(ref, component) {
+  attachRef: function(ref, component, mockConfig) {
     var inst = this.getPublicInstance();
     invariant(inst != null, 'Stateless function components cannot have refs.');
-    var publicComponentInstance = component.getPublicInstance();
+    var publicComponentInstance;
+    if (mockConfig && mockConfig.getMockRef) {
```

Comment:
`getPublicInstance` is already defined inside the test renderer, but I didn't think to pass the `transaction` itself to `getPublicInstance`, which should let us clean up all this domain-specific code 👍